public/javascripts/SVValidate/src/panorama/GSVOverlay.js

```diff
@@ -0,0 +1,143 @@
+        currY: 0,
+        prevX: 0,
+        prevY: 0,
+        leftDownX: 0,
+        leftDownY: 0,
+        leftUpX: 0,
+        leftUpY: 0,
+        isLeftDown: false
+    };
+
+    /**
+     * Disables panning on the GSV window.
+     */
+    function disablePanning() {
+        panningDisabled = true;
+    }
+
+    /**
+     * Enables panning on the GSV window.
+     */
+    function enablePanning() {
+        panningDisabled = false;
+    }
+
+    /**
+     * This is a callback function that is fired with the mouse down event
+     * on the view control layer (where you control street view angle.)
+     * @param e
+     */
+    function handlerViewControlLayerMouseDown (e) {
+        mouseStatus.isLeftDown = true;
+        mouseStatus.leftDownX = mouseposition(e, this).x;
+        mouseStatus.leftDownY = mouseposition(e, this).y;
+        viewControlLayer.css("cursor", "url(/assets/javascripts/SVLabel/img/cursors/closedhand.cur) 4 4, move");
+
+        //This is necessary for supporting touch devices, because there is no mouse hover
+        mouseStatus.prevX = mouseposition(e, this).x;
+        mouseStatus.prevY = mouseposition(e, this).y;
+    }
+
+    /**
+     * This is a callback function that is called with mouse up event on
+     * the view control layer (where you change the Google Street view angle.
+     * @param e
+     */
+    function handlerViewControlLayerMouseUp (e) {
+        viewControlLayer.css("cursor", "url(/assets/javascripts/SVLabel/img/cursors/openhand.cur) 4 4, move");
+        mouseStatus.isLeftDown = false;
+        mouseStatus.leftUpX = mouseposition(e, this).x;
+        mouseStatus.leftUpY = mouseposition(e, this).y;
+
+    }
+    /**
+     * Handles mouse leaving control view.
+     * @param e
+     */
+    function handlerViewControlLayerMouseLeave (e) {
+        viewControlLayer.css("cursor", "url(/assets/javascripts/SVLabel/img/cursors/openhand.cur) 4 4, move");
+        mouseStatus.isLeftDown = false;
+    }
+
+    /**
+     * This is a callback function that is fired when a user moves a mouse on the
+     * view control layer where you change the pov.
+     */
+    function handlerViewControlLayerMouseMove (e) {
+        mouseStatus.currX = mouseposition(e, this).x;
+        mouseStatus.currY = mouseposition(e, this).y;
+
+        var timestamp = new Date().getTime();  // waits till the pano is fully loaded
+        if ((timestamp - svv.panorama.getProperty("prevSetPanoTimestamp") > 800) && mouseStatus.isLeftDown && panningDisabled === false) {
+            // If a mouse is being dragged on the control layer, move the sv image.
+            var dx = mouseStatus.currX - mouseStatus.prevX;
+            var dy = mouseStatus.currY - mouseStatus.prevY;
+            var pov = svv.panorama.getPov();
+            var zoomLevel = pov.zoom;
+            dx = dx / (2 * zoomLevel);
+            dy = dy / (2 * zoomLevel);
+            dx *= 1.5;
+            dy *= 1.5;
+            updatePov(dx, dy);
```

Comment:
Is the code above copied from somewhere else? Or is this code that you just wrote yourself?

Reply:
Most of the code in this file was adapted from MapService.js, which is used to handle users' interactions with the GSV window on the audit page. I modified several things so this could work on /validation.

Reply:
Sounds good!